src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FailOverSupportedConfigurationClient.cs

```diff
@@ -0,0 +1,193 @@
+        {
+            if (clients == null || clients.Count() < 1)
+            {
+                throw new ArgumentNullException(nameof(clients));
+            }
+
+            this.retryPrimaryConfigClientAfter = DateTimeOffset.UtcNow;
+            this.failedRequestsToPrimaryConfigClient = 0;
+            this.clients = clients;
+        }
+
+        public Task<Response<ConfigurationSetting>> GetConfigurationSettingAsync(string key, string label = null, CancellationToken cancellationToken = default)
+        {
+            return ExecuteWithFailOverPolicyAsync(clients.Select((client) => new Func<Task<Response<ConfigurationSetting>>>(() => client.Client.GetConfigurationSettingAsync(key, label, cancellationToken))), cancellationToken);
+        }
+
+        public Task<Response<ConfigurationSetting>> GetConfigurationSettingAsync(ConfigurationSetting setting, bool onlyIfChanged = false, CancellationToken cancellationToken = default)
+        {
+            return ExecuteWithFailOverPolicyAsync(clients.Select((client) => new Func<Task<Response<ConfigurationSetting>>>(() => client.Client.GetConfigurationSettingAsync(setting, onlyIfChanged, cancellationToken))), cancellationToken);
+        }
+
+        public AsyncPageable<ConfigurationSetting> GetConfigurationSettingsAsync(SettingSelector selector, CancellationToken cancellationToken = default)
+        {
+            return ExecuteWithFailOverPolicy(clients.Select((client) => new Func<AsyncPageable<ConfigurationSetting>>(() => client.Client.GetConfigurationSettingsAsync(selector, cancellationToken))), cancellationToken);
+        }
+
+        public void UpdateSyncToken(Uri endpoint, string syncToken)
+        {
+            this.clients.Single(client => client.Endpoint.Host.Equals(endpoint.Host)).Client.UpdateSyncToken(syncToken);
+        }
+
+        private async Task<Response<T>> ExecuteWithFailOverPolicyAsync<T>(IEnumerable<Func<Task<Response<T>>>> delegates, CancellationToken cancellationToken = default)
+        {
+            Exception latestException = null;
+
+            for (var i = ShouldTryPrimaryConfigStore() ? 0 : 1; i < clients.Count(); i++)
+            {
+                var success = false;
+                var operationCanceled = false;
+
+                try
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    var result = await delegates.ElementAt(i)().ConfigureAwait(false);
+                    success = true;
+
+                    return result;
+                }
+                catch (RequestFailedException e) when (e.Status == (int)HttpStatusCode.ServiceUnavailable || e.Status == HttpStatusRequestThrottled)
+                {
+                    latestException = e;
+                    continue;
+                }
+                catch (AggregateException ex)
+                {
+                    if (ex?.InnerExceptions?.All(e => e is RequestFailedException) == true)
```

Comment:
We can move this condition to the `when` clause of this catch block. That would help filter and catch only those `AggregateException`s that we care about.

Reply:
Ideally, that is the optimal way, but the condition is too complicated to put in a `when` block. It would compromise  readability.

Reply:
We already do it in many places in the provider class: https://github.com/Azure/AppConfiguration-DotnetProvider/blob/0a0ab8dc37dcf8681a583cb172f0ed1666c0e6c4/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs#L212

Reply:
We can use exception filters for checking whether all InnerExceptions are RequestFailedException, but checking for the Status of the last exception is not something I would put in exception filters.